src/main/java/com/pubnub/api/managers/RetrofitManager.java

```diff
@@ -124,6 +127,20 @@ public RetrofitManager(PubNub pubNubInstance) {
         this.messageActionService = transactionInstance.create(MessageActionService.class);
         this.filesService = transactionInstance.create(FilesService.class);
         this.s3Service = noSignatureInstance.create(S3Service.class);
+
+        this.pubnub.addListener(new SubscribeCallback.BaseSubscribeCallback() {
+            @Override
+            public void status(@NotNull final PubNub pubnub, @NotNull final PNStatus pnStatus) {
+                if (pnStatus.getCategory() == PNStatusCategory.PNReconnectedCategory) {
+                    Executors.newSingleThreadExecutor().execute(new Runnable() {
+                        @Override
+                        public void run() {
+                            transactionClientInstance.connectionPool().evictAll();
```

Comment:
`transactionClientInstance` gets created only when `!pubNubInstance.getConfiguration().isGoogleAppEngineNetworking()` condition is true.
What happens to this line when `transactionClientInstance` is not created?